crates/bevy_render/src/mesh/primitives/dim3/tetrahedron.rs

```diff
@@ -0,0 +1,55 @@
+use super::triangle3d;
+use crate::{
+    mesh::{Indices, Mesh, Meshable},
+    render_asset::RenderAssetUsages,
+};
+use bevy_math::primitives::{Tetrahedron, Triangle3d};
+use wgpu::PrimitiveTopology;
+
+impl Meshable for Tetrahedron {
+    type Output = Mesh;
+
+    fn mesh(&self) -> Self::Output {
+        let mut faces: Vec<_> = self.faces().into();
+
+        // If the tetrahedron has negative orientation, reverse all the triangles so that
+        // they still face outward.
+        if self.signed_volume().is_sign_negative() {
+            faces.iter_mut().for_each(Triangle3d::reverse);
+        }
+
+        let mut positions = vec![];
+        let mut normals = vec![];
+        let mut uvs = vec![];
+
+        // Each face is meshed as a `Triangle3d`, and we just shove the data into the
+        // vertex attributes sequentially.
+        for face in faces {
+            positions.extend(face.vertices);
+
+            let face_normal = triangle3d::normal_vec(&face);
+            normals.extend(vec![face_normal; 3]);
+
+            let face_uvs = triangle3d::uv_coords(&face);
+            uvs.extend(face_uvs);
+        }
+
+        // There are four faces and none of them share vertices.
+        let indices = Indices::U32(vec![0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]);
```

Comment:
Both! The more obvious one is the first one; in order to get sharp edges, each vertex needs duplicates in order to have all of the normals for faces that meet at that vertex. 

With UVs it's harder to explain but similar; without duplicate vertices, each vertex can only have one UV coordinate, which means the whole thing needs somehow to have global UV coordinates rather than being subdivided into faces. (This kind of thing is much more obvious in the vertex duplication that happens, for example, with `Annulus` — there, you can see that a seam is needed in order to make the UV rectangle wrap into a ring.)

(For certain kinds of tetrahedra, global UV coordinates are probably possible, since you can imagine unfolding the tetrahedron so that its faces all lie flat in some rectangle, but this seems substantially more difficult to do in general.)

Reply:
(Actually I thought about it for a second longer and every tetrahedron can "easily" be unfolded to lie in a plane, but I don't think that UV mapping probably makes a very good default.)